libvast/vast/arrow_extension_types.hpp

```diff
@@ -0,0 +1,52 @@
+#include "vast/fwd.hpp"
+
+#include "vast/type.hpp"
+
+#include <arrow/api.h>
+
+namespace vast {
+
+/// Enum representation in the Arrow type system, utilizing an extension type.
+/// The underlying data is represented as a dictionary, where the `dict` part
+/// contains all the possible variants specified in the underlying VAST enum.
+class enum_extension_type : public arrow::ExtensionType {
+public:
+  explicit enum_extension_type(enumeration_type enum_type)
```

Comment:
API docs for every member function + move to cpp file.

Reply:
clarification: does that include comments for methods that are inherited / override from `arrow::ExtensionType` itself?

Reply:
I think that would make sense, yes. Just a brief single-sentence summary should suffice.